var/spack/repos/builtin/packages/hdfview/package.py

```diff
@@ -29,13 +34,9 @@ class Hdfview(Package):
     patch("fix_build.patch", when="@3.1.1")
 
     depends_on("ant", type="build")
-    depends_on("hdf5 +java")
+    depends_on("hdf5@1.10.9 +java")
     depends_on("hdf +java -external-xdr +shared")
 
-    depends_on("hdf5@1.10", when="@:3.1")
-    depends_on("hdf5@1.12:", when="@3.2")
-    depends_on("hdf5@1.14:", when="@3.3:")
```

Comment:
Based on the fact that you were added `3.1.3` here, I'm assuming that's what you were installing. Can you resstore the last two of these but replace the first:

```
depends_on("hdf5@1.10", when="@:3.1")
```

with 

```
depends_on("hdf5@:1.10.7", when="@3.1.1")
```

(which looks like it would address https://github.com/spack/spack/issues/28236)

and add

```
depends_on("hdf5@:1.10.9", when="@3.1.3")
```

?